### ISSUE TITLE
[Bug] Wallet Balance and Transaction History Missing After Reimporting Wallet on Testnet A

### ISSUE THREAD

**Issue by shahmeerzaidi** [Chia-Network/chia-blockchain#18400]
### What happened?

After deleting my wallet and the corresponding .sqlite file, I restarted the Chia Blockchain and reimported my wallet using the GUI. Despite having a full node synced and being on Testnet A, the wallet shows no transactions and no balance. I attempted to resolve the issue by reinstalling Chia Blockchain and reimporting the wallet multiple times, but the problem persists.

### Steps to Reproduce:

- Delete an existing wallet with balance.
- Navigate to ~/.chia/mainnet/wallet/db and delete the .sqlite file corresponding to the wallet's fingerprint.
- Restart the Chia Blockchain.
- Import the wallet using the GUI:
- Click "Add Wallet" → "Import Existing".
- Enter the wallet name.
- Paste the mnemonic seed phrase.
- Click "Import".
- Observe that the wallet has no transactions and no balance.
- Reinstall Chia Blockchain and repeat the import process.

### Expected Results:

The wallet should display the correct balance and transaction history.

### Actual Results:
The wallet shows no transactions and no balance.

### Environment:

- Chia Blockchain version: 2.4.3-rc1
- Operating System: Ubuntu 22.04.4 LTS
- Network: Testnet A

### Additional Information:

- Full node is fully synced.
- Issue persists despite multiple attempts to reimport the wallet and reinstall Chia Blockchain.
- Using the GUI for importation.
- Logs I provided couldn't be relevant.

### Attachments:

Before:
![Screenshot from 2024-08-05 12-24-54](https://github.com/user-attachments/assets/8f40b8dd-62d7-492f-be20-d8ac1107b4bf)

After:
![image](https://github.com/user-attachments/assets/10a4a43a-a001-4a70-b26b-0d6554bc6d16)

### Version

2.4.3-rc1

### What platform are you using?

Linux

### What ui mode are you using?

GUI

### Relevant log output

```shell
2024-08-05T12:50:43.840 wallet wallet_server              : INFO     Connection closed: 127.0.0.1, node id: 9d4fb753863728e3d213378631918f0b07c79f6c4ebb501eaf1dd3b58ce51b83
2024-08-05T12:50:43.840 full_node full_node_server        : INFO     Connection closed: 127.0.0.1, node id: 4f473aa44126b5a137ca08589441406ee545e1a7179120948a4d4b75a964a88d
2024-08-05T12:50:43.841 full_node chia.full_node.full_node: INFO     peer disconnected PeerInfo(_ip=IPv4Address('127.0.0.1'), _port=0)
2024-08-05T12:50:43.841 wallet chia.server.address_manager_store: INFO     Creating new address_manager
2024-08-05T12:50:44.158 wallet chia.rpc.rpc_server        : WARNING  Error while handling message: Traceback (most recent call last):
  File "chia/rpc/rpc_server.py", line 340, in safe_handle
  File "chia/rpc/rpc_server.py", line 331, in ws_api
  File "chia/rpc/wallet_rpc_api.py", line 1100, in get_transaction_count
  File "chia/wallet/wallet_node.py", line 186, in wallet_state_manager
RuntimeError: wallet state manager not assigned

2024-08-05T12:50:44.159 wallet chia.rpc.rpc_server        : WARNING  Error while handling message: Traceback (most recent call last):
  File "chia/rpc/rpc_server.py", line 340, in safe_handle
  File "chia/rpc/rpc_server.py", line 331, in ws_api
  File "chia/rpc/wallet_rpc_api.py", line 1059, in get_transactions
  File "chia/wallet/wallet_node.py", line 186, in wallet_state_manager
RuntimeError: wallet state manager not assigned

2024-08-05T12:50:44.160 wallet chia.rpc.rpc_server        : WARNING  Error while handling message: Traceback (most recent call last):
  File "chia/rpc/rpc_server.py", line 340, in safe_handle
  File "chia/rpc/rpc_server.py", line 331, in ws_api
  File "chia/rpc/wallet_rpc_api.py", line 993, in get_wallet_balance
  File "chia/rpc/wallet_rpc_api.py", line 975, in _get_wallet_balance
  File "chia/wallet/wallet_node.py", line 186, in wallet_state_manager
RuntimeError: wallet state manager not assigned

2024-08-05T12:50:44.161 wallet chia.rpc.rpc_server        : WARNING  Error while handling message: Traceback (most recent call last):
  File "chia/rpc/rpc_server.py", line 340, in safe_handle
  File "chia/rpc/rpc_server.py", line 331, in ws_api
  File "chia/rpc/wallet_rpc_api.py", line 1059, in get_transactions
  File "chia/wallet/wallet_node.py", line 186, in wallet_state_manager
RuntimeError: wallet state manager not assigned

2024-08-05T12:50:46.064 wallet wallet_server              : INFO     Connected with full_node PeerInfo(_ip=IPv4Address('127.0.0.1'), _port=58444)
2024-08-05T12:50:46.068 wallet wallet_server              : WARNING  API not ready, ignore request: {'data': '0x6c5be2d9346e4501f1d30fa953041ec50ea36651066d2c4e7bb706f5d8751d1e00199145000000000000000000000001d6c9be2f00199145',
 'id': None,
 'type': 50}
2024-08-05T12:50:48.222 wallet chia.rpc.rpc_server        : WARNING  Error while handling message: Traceback (most recent call last):
  File "chia/rpc/rpc_server.py", line 340, in safe_handle
  File "chia/rpc/rpc_server.py", line 331, in ws_api
  File "chia/rpc/wallet_rpc_api.py", line 602, in get_sync_status
  File "chia/wallet/wallet_node.py", line 186, in wallet_state_manager
RuntimeError: wallet state manager not assigned

2024-08-05T12:50:49.011 full_node chia.full_node.full_node: INFO     ⏲️  Finished signage point 40/64: CC: cb792585b6c47c5f04c5b9b43c34441405b70081ed2c204b8c5cf7ca103699e5 RC: cb386cb005c596ff4d4854e11f18a9ecd9aee49df3df09a9202e94dd956435d8 
2024-08-05T12:50:49.013 farmer farmer_server              : WARNING  API not ready, ignore request: {'data': '0x84506bb5ea436caa8b2fc0d15f2f6384855cfba165b99bb8176a89e2f906d3e6cb792585b6c47c5f04c5b9b43c34441405b70081ed2c204b8c5cf7ca103699e5cb386cb005c596ff4d4854e11f18a9ecd9aee49df3df09a9202e94dd956435d80000000000000d20000000000418000028001991450100010000abf0a0fcc6fd0d66b8a710bc83e9cf2b0a4b2c5c217584af3d56597494a6e57138773b9e9dfb788cd7ad82957c0b7d31f48602635fb67d22c766cab7ae113c127be5918592fdcdab2612293bbbcc3858b000003b334e0a2b7037781e76afd22b01000000616c98ff0f4e93e47731c9b82cb297630a84d98f32904007c6dc07d59eec99457b56612bc7a072cae24d08c07ef08a0ea60d5d2df9e9cf138b8998e15684ea0d36a598b41a24f7984cf01feac7b6d51d627e69359ed9cc871efab2c065cfd6020201',
 'id': None,
 'type': 8}
2024-08-05T12:50:56.886 full_node chia.full_node.full_node: INFO     ⏲️  Finished signage point 41/64: CC: 62db222454709c6e02fcb668041c652fd397abb00181721e0590f6a9206d026b RC: 584f68dad037dd3dab3ca16c210b30c25670a00e6b6c400e54a42e6d8ee7932a 
2024-08-05T12:50:56.888 farmer farmer_server              : WARNING  API not ready, ignore request: {'data': '0x84506bb5ea436caa8b2fc0d15f2f6384855cfba165b99bb8176a89e2f906d3e662db222454709c6e02fcb668041c652fd397abb00181721e0590f6a9206d026b584f68dad037dd3dab3ca16c210b30c25670a00e6b6c400e54a42e6d8ee7932a0000000000000d20000000000418000029001991450100010200767c87b0b5cf2711064888ee94931c31f4b145b431cd296917e92f9a6cb39cdd12e71efa68e6531caeb32e9de5abd8ace57c0a83b472a4c1fab678d17552d548c9057bc9a9d960b5f53fabe3f113982a9ae5afb9ae21d253734f0da94c54c66c01000300937844183ab740898a305f365a70c5ba1eb10dea8903a15f9a2b8d6d9ba0179d0b628bd740c944112197be42d37a8329757fad348fc86a5ba25451e92f3892201689bff6e6ec51d858d46ef58c94b605533c6922205509a67b0a436db7a151020100',
 'id': None,
 'type': 8}
2024-08-05T12:50:58.235 wallet chia.rpc.rpc_server        : WARNING  Error while handling message: Traceback (most recent call last):
  File "chia/rpc/rpc_server.py", line 340, in safe_handle
  File "chia/rpc/rpc_server.py", line 331, in ws_api
  File "chia/rpc/wallet_rpc_api.py", line 602, in get_sync_status
  File "chia/wallet/wallet_node.py", line 186, in wallet_state_manager
RuntimeError: wallet state manager not assigned

2024-08-05T12:51:04.988 full_node chia.full_node.full_node: INFO     ⏲️  Finished signage point 42/64: CC: a5830f027c415469461aefde0e79083dbe43e35d1d303481aa53b06f473769b2 RC: c704ea9eab1b7a3f20462e5dd21743a3516b0d7281c07720f057c7078ac85678 
2024-08-05T12:51:04.989 farmer farmer_server              : WARNING  API not ready, ignore request: {'data': '0x84506bb5ea436caa8b2fc0d15f2f6384855cfba165b99bb8176a89e2f906d3e6a5830f027c415469461aefde0e79083dbe43e35d1d303481aa53b06f473769b2c704ea9eab1b7a3f20462e5dd21743a3516b0d7281c07720f057c7078ac856780000000000000d2000000000041800002a0019914501000100007b516e3fcf5d0a3fc32589ad3158f73945203317456c6eda13bc503ad34370abb62a0d3dbda87ad0e5f6590cd33d0dd15b8c82ac7fe836b50a5d416ad111cc24eda1f04563ff662c4fc0ee828310466bf48ad088db30907bc0fbf5a4461e212b010000008a59353a84173a152e0d99840d0017dba6ff423f0cecd53428e1e4f80d99f03fa3db83afdd2324367dcfe1d7d8eb4ba23832c43957e977f17d94ba84ea8e676dd3f805cc0733cf6e377e76c95311b57e36b1ffa5a663153be8b75d59484d485a0100',
 'id': None,
 'type': 8}
2024-08-05T12:51:08.311 wallet chia.wallet.wallet_node    : INFO     self._close
2024-08-05T12:51:08.312 wallet chia.wallet.wallet_node    : INFO     self._await_closed
2024-08-05T12:51:08.312 wallet wallet_server              : INFO     Connection closed: 127.0.0.1, node id: 9d4fb753863728e3d213378631918f0b07c79f6c4ebb501eaf1dd3b58ce51b83
2024-08-05T12:51:08.313 full_node full_node_server        : INFO     Connection closed: 127.0.0.1, node id: 4f473aa44126b5a137ca08589441406ee545e1a7179120948a4d4b75a964a88d
2024-08-05T12:51:08.313 full_node chia.full_node.full_node: INFO     peer disconnected PeerInfo(_ip=IPv4Address('127.0.0.1'), _port=0)
2024-08-05T12:51:08.314 wallet wallet_server              : ERROR    Caught exception: RuntimeError: wallet state manager not assigned
Traceback (most recent call last):
  File "chia/util/log_exceptions.py", line 20, in log_exceptions
  File "chia/server/ws_connection.py", line 344, in close
  File "chia/server/server.py", line 559, in connection_closed
  File "chia/wallet/wallet_node.py", line 745, in on_disconnect
  File "chia/wallet/wallet_node.py", line 186, in wallet_state_manager
RuntimeError: wallet state manager not assigned

2024-08-05T12:51:08.314 wallet chia.wallet.wallet_node    : INFO     Selected multiprocessing start method: default
2024-08-05T12:51:08.316 wallet chia.server.address_manager_store: INFO     Creating new address_manager
2024-08-05T12:51:09.137 wallet chia.wallet.wallet_node    : INFO     Wallet is logged in using key with fingerprint: 541826640
2024-08-05T12:51:09.137 wallet chia.wallet.wallet_node    : INFO     Updated last used fingerprint: 541826640
2024-08-05T12:51:09.190 wallet chia.server.address_manager_store: INFO     Creating new address_manager
2024-08-05T12:51:10.103 wallet wallet_server              : INFO     Connected with full_node PeerInfo(_ip=IPv4Address('127.0.0.1'), _port=58444)
2024-08-05T12:51:10.128 wallet chia.wallet.wallet_state_manager: INFO     set_sync_mode syncing - range: 1675573-1675589
2024-08-05T12:51:10.133 wallet chia.wallet.wallet_blockchain: INFO     Peak set to: 1675589 timestamp: 1722844202
2024-08-05T12:51:10.134 wallet chia.wallet.wallet_node    : INFO     Starting sync trusted: True to peer 127.0.0.1
2024-08-05T12:51:10.136 wallet chia.wallet.wallet_node    : INFO     perform_atomic_rollback to 1675317
```

**Comment by BrandtH22** [comment]
Hey @shahmeerzaidi , did you have any other wallets on the same machine when performing these steps? 
I am wondering if this is related to #18396 . Essentially if you have another wallet that has no transactions then this issue might already be resolved with #18405

**Comment by BrandtH22** [comment]
Hey @shahmeerzaidi , are you able to reproduce this issue on rc3?

Also when reproducing please first ensure that you have deleted the wallet dbs prior to syncing with rc3 to ensure there are no left over issues from the rc1 bug. Let me know if you have any questions and thank you for the information!
